hal/src/main/native/athena/DIO.cpp

```diff
@@ -126,6 +126,9 @@ void HAL_FreeDIOPort(HAL_DigitalHandle dioPortHandle) {
   // no status, so no need to check for a proper free.
   digitalChannelHandles->Free(dioPortHandle, HAL_HandleEnum::DIO);
   if (port == nullptr) return;
+  // Wait for no other object to hold this handle.
+  while (port.use_count() != 1) {
+  }
```

Comment:
Put `std::this_thread::yield();` into the loop body.

Reply:
Should I spin a few times before doing this? In most cases this won't get hit, and we can avoid the context switch? Or not worth it?

Reply:
Measure it?

Reply:
Measure what? A yield is going to do a context switch no matter what. But it doesn't get hit often so it's probably fine.

Reply:
Since this is on a delete and its a race condition that would cause  the context switch anyway, I just put it in. Trying to induce the actual condition is probably difficult.